graylog-storage-elasticsearch6/src/main/java/org/graylog/storage/elasticsearch6/views/ElasticsearchBackend.java

```diff
@@ -84,109 +81,91 @@ public class ElasticsearchBackend implements QueryBackend<ESGeneratedQueryContex
         this.elasticsearchSearchTypeHandlers = elasticsearchSearchTypeHandlers;
         this.jestClient = jestClient;
         this.indexLookup = indexLookup;
 
-        this.queryStringDecorators = queryStringDecorators;
         this.queryContextFactory = queryContextFactory;
         this.usedSearchFiltersToQueryStringsMapper = usedSearchFiltersToQueryStringsMapper;
         this.allowLeadingWildcard = allowLeadingWildcard;
     }
 
-    private QueryBuilder normalizeQueryString(String queryString) {
+    private QueryBuilder translateQueryString(String queryString) {
         return (queryString.isEmpty() || queryString.trim().equals("*"))
                 ? QueryBuilders.matchAllQuery()
                 : QueryBuilders.queryStringQuery(queryString).allowLeadingWildcard(allowLeadingWildcard);
     }
 
     @Override
-    public ESGeneratedQueryContext generate(SearchJob job, Query query, SearchConfig searchConfig) {
+    public ESGeneratedQueryContext generate(SearchJob job, Query query, Set<SearchError> validationErrors) {
         final BackendQuery backendQuery = query.query();
 
-        validateQueryTimeRange(query, searchConfig);
-
         final Set<SearchType> searchTypes = query.searchTypes();
 
-        final String queryString = this.queryStringDecorators.decorate(backendQuery.queryString(), job, query);
-        final QueryBuilder normalizedRootQuery = normalizeQueryString(queryString);
+        final QueryBuilder normalizedRootQuery = translateQueryString(backendQuery.queryString());
 
         final BoolQueryBuilder boolQuery = QueryBuilders.boolQuery()
                 .filter(normalizedRootQuery);
 
         usedSearchFiltersToQueryStringsMapper.map(query.filters())
                 .stream()
-                .map(searchFilterQueryString -> this.queryStringDecorators.decorate(searchFilterQueryString, job, query))
-                .map(this::normalizeQueryString)
+                .map(this::translateQueryString)
                 .forEach(boolQuery::filter);
 
         // add the optional root query filters
         generateFilterClause(query.filter(), job, query)
-                .map(boolQuery::filter);
+                .ifPresent(boolQuery::filter);
 
         final SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder()
                 .query(boolQuery)
                 .from(0)
                 .size(0);
 
-        final ESGeneratedQueryContext queryContext = queryContextFactory.create(this, searchSourceBuilder, job, query);
-        for (SearchType searchType : searchTypes) {
-
-            final Optional<SearchTypeError> searchTypeError = validateSearchType(query, searchType, searchConfig);
-            if(searchTypeError.isPresent()) {
-                LOG.error("Invalid search type {} for elasticsearch backend, cannot generate query part. Skipping this search type.", searchType.type());
-                queryContext.addError(searchTypeError.get());
```

Comment:
This part is gone after change in lines 128-130, introducing filtered streams. Search Types with error seem to be simply filtered out now. If it is intentional, please disregard my comment, but if we want to keep the old behavior, we may need to bring it back.